syntax = "proto3";

package master_server;

service MasterServer {
  rpc GetDistribution (GetDistributionRequest) returns (GetDistributionResponse) {}
}

message GetDistributionRequest {
  string filename = 1;
  int64 beginFrame = 2;
  optional int64 endFrame = 3;
}

message FrameDistribution {
  string endpoint = 1;
  int64 beginFrame = 2;
  int64 endFrame = 3;
}

message GetDistributionResponse {
  bool endOfFile = 1;
  repeated FrameDistribution distribution = 2;
}
